src/libmongoc/tests/test-mongoc-collection.c

```diff
@@ -2621,6 +2621,18 @@ test_estimated_document_count (void)
    ASSERT_OR_PRINT (123 == future_get_int64_t (future), error);
    ASSERT_MATCH (&reply, server_reply);
 
+   future_destroy (future);
+
+   /* CDRIVER-3612: ensure that an explicit session triggers a client error */
+   future = future_collection_estimated_document_count (
```

Comment:
Since the mock server is not contacted, you can call `mongoc_collection_estimated_document_count` directly.

Reply:
My preference is leave it this way to keep the test pattern uniform.